### ISSUE TITLE
[Feature]: support generate overview page by flatten configs

### ISSUE THREAD

**Issue by 9aoy** [web-infra-dev/rspress#190]
### What problem does this feature solve?

expect generate overview page by below: 

<img width="1077" alt="image" src="https://github.com/web-infra-dev/rspress/assets/22373761/1cbbdcc3-931c-44f8-bcac-e4be221e5e8a">

<img width="1407" alt="image" src="https://github.com/web-infra-dev/rspress/assets/22373761/d0640c8c-6b51-4f08-bf27-1ddcb83ec5e6">

<img width="383" alt="image" src="https://github.com/web-infra-dev/rspress/assets/22373761/dd8a23db-0d69-4cfb-a501-3fd26e2d87b8">

### What does the proposed API look like?

none

**Comment by sanyuan0704** [comment]
Is there any function that rspress hasn't implemented?

**Comment by 9aoy** [comment]
Is this already supported? Let me try it.

**Comment by sanyuan0704** [comment]
No, i don't understand what you mean. Currrent structure in config directory can be rendered normally.
